src/Data/Detection/Common/DetectionFactory/DetectionFactory.cs

```diff
@@ -0,0 +1,15 @@
+namespace Data.Detection
```

Comment:
Namespace does not match the folder structure. Should we make sure we don't violate this rule?

Reply:
Done